src/notebook/plugin.ts

```diff
@@ -143,19 +172,26 @@ class NotebookFileHandler extends AbstractFileHandler {
     let model = new NotebookModel();
     let panel = new Panel();
 
-    let button = new Widget();
-    let b = document.createElement('button');
-    b.appendChild(document.createTextNode('Execute Current Cell'))
-    button.node.appendChild(b);
-
-
     let widgetarea = new Widget();
     let manager = new WidgetManager(widgetarea.node);
 
+    let notebookId = this.notebookId;
+    this.notebookId += 1;
+
     this.session.startNew({notebookPath: contents.path}).then(s => {
-      b.addEventListener('click', ev=> {
-        executeSelectedCell(model, s);
-      })
+      // TODO: it's probably better to make *one* shortcut that executes whatever
+      // the current notebook's selected cell is, rather than registering a
+      // a new shortcut for every open notebook.  To do this, we need the application
+      // state (or somewhere) to include the 'current document'
+      // this is better because it makes one shortcut, and it also is better because
+      // then it is easy for the command to be invoked via the menu, palette, plugin, etc.
```

Comment:
Yeah, I think it should be separate from the opener, at a top level
application state thing.

On Fri, Jan 29, 2016, 18:32 Steven Silvester <notifications@github.com>
wrote:

> In src/notebook/plugin.ts
> <https://github.com/jupyter/jupyter-js-plugins/pull/50#discussion_r51330343>
> :
>
> >      this.session.startNew({notebookPath: contents.path}).then(s => {
> > -      b.addEventListener('click', ev=> {
> > -        executeSelectedCell(model, s);
> > -      })
> > +      // TODO: it's probably better to make *one* shortcut that executes whatever
> > +      // the current notebook's selected cell is, rather than registering a
> > +      // a new shortcut for every open notebook.  To do this, we need the application
> > +      // state (or somewhere) to include the 'current document'
> > +      // this is better because it makes one shortcut, and it also is better because
> > +      // then it is easy for the command to be invoked via the menu, palette, plugin, etc.
>
> Another alternative is a plugin that listens to document level focus and
> blur events and maintains the last active element:
> http://stackoverflow.com/a/28194767
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter-js-plugins/pull/50/files#r51330343>.
>